lib/compress/zstd_lazy.c

```diff
@@ -865,204 +865,32 @@ FORCE_INLINE_TEMPLATE size_t ZSTD_HcFindBestMatch_extDict_selectMLS (
 * (SIMD) Row-based matchfinder
 ***********************************/
 /* Constants for row-based hash */
-#define ZSTD_ROW_HASH_TAG_OFFSET 1                               /* byte offset of hashes in the match state's tagTable from the beginning of a row */
+#define ZSTD_ROW_HASH_TAG_OFFSET 16                              /* byte offset of hashes in the match state's tagTable from the beginning of a row */
```

Comment:
I've done some quick benchmarks with 16-byte aligned reads against unaligned - unfortunately I didn't measure any performance benefit.

Reply:
AFAIK  --- For this particular access pattern, aligned loads are pointless on modern CPUs.  Though, who knows what strange CPUs will use the scalar path (ARMv5 or below)? If I understand correctly, an AMD Jaguar (PS4 or Xbox One) using an align 16-byte load at a 16-byte boundary is 3 cycles faster.